### PR TITLE
SONARJS-676 Improve issue location for illegal expression rules

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/CommaOperatorUseCheck.java [hunk 1]
```diff
@@ -55,14 +58,16 @@ public void visitBinaryExpression(BinaryExpressionTree tree) {
...
+    getAllSubExpressions(tree).forEach(expression -> super.scan(expression));
+  }
 
-    for (ExpressionTree expression : expressions) {
-      super.scan(expression);
-    }
+  private void raiseIssue(BinaryExpressionTree tree) {
+    List<SyntaxToken> commas = getCommas(tree);
+    String message = commas.size() > 1 ? MESSAGE_MANY_COMMAS : MESSAGE_ONE_COMMA;
+    PreciseIssue issue = addIssue(commas.get(0), message);
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/CommaOperatorUseCheck.java]: MAJOR Remove this useless assignment to local variable "issue". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1854)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [NullDereferenceInConditionalCheck.java#L91](https://github.com/SonarSource/sonar-javascript/blob/b35a13db42a438186f33ff5e247ba931bfa6c55b/javascript-checks/src/main/java/org/sonar/javascript/checks/NullDereferenceInConditionalCheck.java#L91): Make this a "static" inner class. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ComparisonWithNaNCheck.java [hunk 2]
```diff
@@ -57,15 +58,28 @@
...
+      addIssue(nan, String.format(MESSAGE, expression.operator().text()))
+        .secondary(expression.operator());
     }
   }
 
   private static boolean isNaN(ExpressionTree expression) {
     return expression.is(Kind.IDENTIFIER_REFERENCE) && "NaN".equals(((IdentifierTree) expression).name());
   }
 
+  @Nullable
```
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ComparisonWithNaNCheck.java]: @vilchik-elena I'm not sure of the semantic difference between `@Nullable` and `@CheckForNull`, but the SonarQube java rule on NPE checks how method return values are used only for methods annotated with `@CheckForNull`.

Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ComparisonWithNaNCheck.java]: @pynicolas thanks! fixed 2098beed4ff41492bcf613f5f2dd0e46897dca1b

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [NullDereferenceInConditionalCheck.java#L91](https://github.com/SonarSource/sonar-javascript/blob/2098beed4ff41492bcf613f5f2dd0e46897dca1b/javascript-checks/src/main/java/org/sonar/javascript/checks/NullDereferenceInConditionalCheck.java#L91): Make this a "static" inner class. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

**Pull request merged**